File: app/src/main/java/ru/eugen/noteseugen/data/CardDataMapping.java
package ru.eugen.noteseugen.data;

import com.google.firebase.Timestamp;

import java.util.HashMap;
import java.util.Map;

public class CardDataMapping {
    public static class Fields{
        public final static String DATE = "date";
        public final static String NOTE = "note";
        public final static String ESSENCE = "essence";
    }
    public static Card toCard(String id, Map<String, Object> doc) {
        Timestamp timeStamp = (Timestamp)doc.get(Fields.DATE);
        Card card = new Card((String) doc.get(Fields.NOTE),
                (String) doc.get(Fields.ESSENCE),
                timeStamp.toDate());
        card.setId(id);
        return card;
    }
    public static Map<String, Object> toDocument(Card card){
        Map<String, Object> doc = new HashMap<>();
        doc.put(Fields.NOTE, card.getNote());
        doc.put(Fields.ESSENCE, card.getEssence());
        doc.put(Fields.DATE, card.getDate());
        return doc;
    }
}

File: app/src/main/java/ru/eugen/noteseugen/data/CardDataMapping.kt
package ru.eugen.noteseugen.data;

import com.google.firebase.Timestamp;

import java.util.HashMap;
import java.util.Map;

class CardDataMapping {
    object Fields{
        const val DATE:String  = "date"
        const val NOTE: String  = "note"
        const val ESSENCE: String  = "essence"
    }

    companion object{

        @JvmStatic
        fun toCard(id2:String , doc:MutableMap<String, Any> ) :Card {
            val timeStamp: Timestamp  = doc.get(Fields.DATE) as Timestamp
            val card: Card = Card(doc.get(Fields.NOTE) as String,
                    doc.get(Fields.ESSENCE) as String,
                    timeStamp.toDate());
            card.id = id2
            return card
        }
        @JvmStatic
        fun  toDocument(card:Card) : MutableMap<String, Any> {
            val doc:MutableMap<String, Any>  = HashMap()
            doc.put(Fields.NOTE, card.note)
            doc.put(Fields.ESSENCE, card.essence)
            doc.put(Fields.DATE, card.date)
            return doc;
        }
    }


}
